src/components/common/SmartPrice.tsx

```diff
@@ -0,0 +1,65 @@
+import { calculatePrice, formatPrice, invertPrice, safeTokenName } from '@gnosis.pm/dex-js'
+import React, { useMemo, useState } from 'react'
+import BN from 'bn.js'
+import { TokenDetails } from 'types'
+
+import { SwapPrice } from 'components/common/SwapPrice'
+import { getMarket } from 'utils'
+
+interface Fraction {
+  denominator: BN
+  numerator: BN
+}
+
+interface Props {
+  buyToken: TokenDetails
+  sellToken: TokenDetails
+  price: Fraction
+}
+
+export const SmartPrice: React.FC<Props> = ({ buyToken, sellToken, price: priceFraction }) => {
+  const [isPriceInverted, setIsPriceInverted] = useState(false)
+  const { baseToken, quoteToken } = getMarket({ sellToken, receiveToken: buyToken })
```

Comment:
Does `getMarket` return const refs for tokens? Maybe also memoize to have const deps in `useMemo`

Reply:
this is why I prefered `getMarket` as a hook, but can be easily combined now to make one using the `util` function and a hook

Reply:
Not sure I understand 100%

It’s not always clear to me what we expect to memoize and what not.
In this case, I thought about memoizing, but then I thought it would be a fast computation, so not sure if it would be worth it.
the function just return the original objects (tells u what is the base/quote) so, instances don’t change.
Where do you see the line in what to memoize and what not? Why a hook here is helpful? Is it to allow to reuse the util+memo logic in other parts of the app?

Reply:
Yes you're right that in this case it's probably fine. "Expensive" calculation is a subjective term in some regards and i think this function is relatively simple enough.